netbout-web/src/main/java/com/netbout/email/EmCatch.java

```diff
@@ -0,0 +1,186 @@
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+ * OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+package com.netbout.email;
+
+import com.jcabi.aspects.Immutable;
+import com.jcabi.aspects.Loggable;
+import com.jcabi.log.Logger;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import javax.mail.Flags;
+import javax.mail.Folder;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.NoSuchProviderException;
+import javax.mail.Session;
+import javax.mail.Store;
+import javax.mail.URLName;
+import javax.mail.search.FlagTerm;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+/**
+ * Email monitors an email account periodically
+ * through a daemon thread.
+ * @author Erim Erturk (erimerturk@gmail.com)
+ * @version $Id$
+ * @since 2.15
+ */
+@Immutable
+@Loggable(Loggable.DEBUG)
+@ToString(of = { "action", "user", "password" })
+@EqualsAndHashCode(of = { "action", "user", "password" })
+final class EmCatch {
+    /**
+     * Max sleep time of thread.
+     */
+    private static final long MAX_SLEEP_TIME = 24 * 60 * 60 * 1000;
+    /**
+     * Action.
+     */
+    private final transient Action action;
+    /**
+     * Email Login key.
+     */
+    private final transient String user;
+    /**
+     * Email password.
+     */
+    private final transient String password;
+    /**
+     * Email server host.
+     */
+    private final transient String host;
+    /**
+     * Email server port.
+     */
+    private final transient int port;
+    /**
+     * Ctor.
+     * @param act Email message handler
+     * @param usr Email login key
+     * @param pass Email password
+     * @param hst Email server host
+     * @param prt Email server port
+     * @param prd Email server check period in milliseconds
+     * @checkstyle ParameterNumberCheck (3 lines)
+     */
+    @SuppressWarnings("PMD.DoNotUseThreads")
+    EmCatch(final Action act, final String usr, final String pass,
+            final String hst, final int prt, final long prd) {
+        this.action = act;
+        this.user = usr;
+        this.password = pass;
+        this.host = hst;
+        this.port = prt;
+        final Thread monitor = new Thread(
+            new Runnable() {
+                @Override
+                public void run() {
+                    long period = prd;
+                    while (true) {
+                        try {
+                            EmCatch.this.check();
+                            TimeUnit.SECONDS.sleep(period);
+                            period = prd;
+                        // @checkstyle LineLength (1 lines)
+                        } catch (final InterruptedException | MessagingException ex) {
```

Comment:
@erimerturk we should keep the catch for `InterruptedException` separate in my opinion. In that case I believe the thread should stop.